plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -278,7 +278,14 @@ public void createTable(Table table, PrincipalPrivileges principalPrivileges)
     @Override
     public void dropTable(String databaseName, String tableName, boolean deleteData)
     {
-        delegate.dropTable(databaseName, tableName, deleteData);
+        try {
+            delegate.dropTable(databaseName, tableName, deleteData);
+        }
+        catch (TableNotFoundException e) {
+            // Checking for table existence is done in HiveMetadata, so we can ignore this exception.
```

Comment:
note that the handling in io.trino.plugin.hive.TrinoViewHiveMetastore#dropView for `TableNotFoundException` is now rendered useless.